runtime/compiler/p/codegen/J9TreeEvaluator.cpp

```diff
@@ -12693,6 +13615,588 @@ TR::Register *J9::Power::TreeEvaluator::tabortEvaluator(TR::Node *node, TR::Code
+         generateConditionalBranchInstruction(cg, TR::InstOpCode::bne, node, helperLabel, condReg);
+
+         // Generate assembly for inlined version of array copy.
+         inlineArrayCopy_ICF(node, len, srcAddrReg, dstAddrReg, cg, condReg,
+                             tmp1Reg, tmp2Reg, tmp3Reg, tmp4Reg,
+                             fp1Reg, fp2Reg, fp3Reg, fp4Reg);
+
+         generateLabelInstruction(cg, TR::InstOpCode::b, node, endLabel);
+
+         // Start of referenceArrayCopy helper path.
+         generateLabelInstruction(cg, TR::InstOpCode::label, node, helperLabel);
+
+         TR::PPCJNILinkage *jniLinkage = (TR::PPCJNILinkage*) cg->getLinkage(TR_J9JNILinkage);
+         const TR::PPCLinkageProperties &pp = jniLinkage->getProperties();
+
+         int32_t elementSize;
+         if (comp->useCompressedPointers())
+            elementSize = TR::Compiler->om.sizeofReferenceField();
+         else
+            elementSize = (int32_t) TR::Compiler->om.sizeofReferenceAddress();
+
+         // Sign extend non-64bit Integers on LinuxPPC64 as required by the ABI
+         if (TR::Compiler->target.isLinux() && TR::Compiler->target.is64Bit())
+            {
+            generateTrg1Src1Instruction(cg, TR::InstOpCode::extsw, node, lengthReg, lengthReg);
+            }
+
+         // The C routine expects length measured by slots.
+         generateShiftRightLogicalImmediate(cg, node, lengthReg, lengthReg, trailingZeroes(elementSize));
+
+         generateTrg1Src1Instruction(cg, TR::InstOpCode::mr, node, r3Reg, metaReg);
+
+         TR::RegisterDependencyConditions *helperDeps = new (cg->trHeapMemory()) TR::RegisterDependencyConditions(0, 0, cg->trMemory());
+         TR::SymbolReference *helperSym = comp->getSymRefTab()->findOrCreateRuntimeHelper(TR_referenceArrayCopy, false, false, false);
+
+         gcPoint = generateDepImmSymInstruction(cg, TR::InstOpCode::bl, node, (uintptrj_t)helperSym->getMethodAddress(), helperDeps, helperSym);
+         gcPoint->PPCNeedsGCMap(pp.getPreservedRegisterMapForGC());
+
+         generateDepLabelInstruction(cg, TR::InstOpCode::label, node, endLabel, deps);
+
+         TR::TreeEvaluator::genWrtbarForArrayCopy(node, srcObjReg, dstObjReg, cg);
+
+         cg->decReferenceCount(srcObjNode);
+         cg->decReferenceCount(dstObjNode);
+         cg->decReferenceCount(srcAddrNode);
+         cg->decReferenceCount(dstAddrNode);
+         cg->decReferenceCount(lengthNode);
+
+         if (stopUsingCopyReg1)
+            cg->stopUsingRegister(srcObjReg);
+         if (stopUsingCopyReg2)
+            cg->stopUsingRegister(dstObjReg);
+         if (stopUsingCopyReg3)
+            cg->stopUsingRegister(srcAddrReg);
+         if (stopUsingCopyReg4)
+            cg->stopUsingRegister(dstAddrReg);
+         if (stopUsingCopyReg5)
+            cg->stopUsingRegister(lengthReg);
+
+         if (tmp1Reg)
+            cg->stopUsingRegister(tmp1Reg);
+         if (tmp2Reg)
+            cg->stopUsingRegister(tmp2Reg);
```

Comment:
All of these NULL checks are redundant. `stopUsingRegister` already performs the NULL check for us so this is extra path length which is not needed. Also makes the code much more verbose!

Reply:
Similarly elsewhere in this entire PR.

Reply:
Thanks. @IBMJimmyk could you please double check that stopUsingRegister() checks for NULL and remove redundant if's?

Reply:
I can confirm that stopUsingRegister already performs the NULL check so the extra NULL checks can be removed.